code-scanning/policy-validator-cfn.yaml

```
@@ -0,0 +1,115 @@
+# 2. Copy this workflow and add it under .github/workflows folder of your GitHub repository with name `policy-validator-iam-policies-cfn.yaml`
+# 3. If you're using CheckNoNewAccess check, you need to create a reference policy. Use the guide [here](https://github.com/aws-samples/iam-access-analyzer-custom-policy-check-samples?tab=readme-ov-file#how-do-i-write-my-own-reference-policies)and store them in S3 bucket / GitHub secrets to compare them against the policies in the CFN templates. In below workflow, we are storing the reference policy in the GitHub secrets with name `REFERENCE_IDENTITY_POLICY`
+# 4. If you're using the CheckAccessNotGranted check, identify the critical actions that shouldn't be granted access by the policies in the CFN templates. Store these actions in S3 bucket / GitHub secrets to compare them against the policies in the CFN templates. In the below workflow, we are storing the S3 bucket object containing the critical action in the GitHub secret with name `CRITICAL_ACTIONS`
+# 5. Create a new workflow under ./github/workflows and refer this workflow. Configure the workflow with events to run and path to the CFN templates to be validated. Reference sample code:
+# ```
+# name: Policy Validator for AWS IAM policies in CloudFormation templates
+# on:
+#   pull_request:
+#     types: [opened, review_requested]
+#   push:
+#     branches:
+#       - 'main'
+# jobs:
+#     security-scanner-developers:
+#         uses: .github/workflows/policy-validator-iam-policies-cfn.yaml@main
+#         secrets: inherit
+#         with:
+#             template-path: file-path-to-cfn-template.yaml
+#             region: us-west-2
+# ```
+# 4. Start using the GitHub actions by generating the GitHub events matching the defined criteria in your workflow.
+name: Re-usable workflow for validating AWS IAM policies in CloudFormation templates using Policy Validator
+on:
+  # This workflow is written to illustrate it to use it as re-usable workflow.
+  workflow_call:
+    inputs:
+      template-path:
+        required: true
+        type: string
+      region:
+        required: true
+        type: string
+    outputs:
+      result:
+        value: string
+  # Uncomment the below lines if you want to run this workflow as against push / pull request of default branch
+  # Note: Please pass the value of the inputs in the below workflow by replacing ${{ inputs.<variable> }} with value if you don't want to use it as a re-usable workflow
+  # push:
+  #   branches: [$default-branch, $protected-branches]
+  # pull_request:
+  #   # The branches below must be a subset of the branches above
+  #   branches: [$default-branch]
```

Modifying this to be opt-out vs opt-in per the PR style guide:

```suggestion
  # Note: Please pass the value of the inputs in the below workflow by replacing ${{ inputs.<variable> }} with value if you don't want to use it as a re-usable workflow
  push:
    branches: [$default-branch, $protected-branches]
  pull_request:
    # The branches below must be a subset of the branches above
    branches: [$default-branch]
```

---

Hi @alexisabril, thanks for your feedback. Quick question-by adding qualifying events for `push` and `pull_request`, customers must update the workflow steps with the inputs, as now the inputs will not be coming from caller workflow. How do you recommend us to surface it?

The idea of this workflow is to demonstrate customers how to use these checks in a re-usable workflow and call them from caller workflow.

---

This is an interesting question! My personal perspective is that Starter Workflows would be defining the workflow to be used for a given run. A re-usable workflow would be a workflow that lives elsewhere, such as in a repo a run can access, but not defined in the running repo itself.

As Starter Workflows are intended to pull a workflow into the repo itself, the "re-usable" piece becomes a bit nested. Might I ask, why not use the default workflow here, while sharing the re-usable workflow from the Action repo itself?

---

Hi @alexisabril - Yeah, that's an option we definitely can consider. Let me check with my team and get back to you. 

Meanwhile, I already pushed a change to make the current workflow a default standalone workflow. Can you review it when you get a chance? Thank you!